### PR TITLE
Add search use case (no view)

### PR DESCRIPTION
**Pull request by hydrabeer**
Completed the application business rules and interface adapter layers for the search use case

### CODE REVIEW

File: src/main/java/service/search/interface_adapter/SearchViewModel.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import interface_adapter.ViewModel;
+
+/**
+ * The search view model.
+ */
+public class SearchViewModel extends ViewModel<SearchState> {
+
+    public SearchViewModel() {
+        super("search");
```
Comment (bbayazit16) [inline src/main/java/service/search/interface_adapter/SearchViewModel.java]: Could you factor this into ViewConstants?

**Comment by bbayazit16** [comment]
It'd be great if "search" can be factored into ViewConstants. Other than that, the code looks good, thank you!

**Review by bbayazit16 — APPROVED** [review]
Thanks!

File: src/main/java/service/search/SearchInteractor.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        for (AbstractVaultItem item : searchDAO.getItems()) {
+            if (item.getTitle().toLowerCase().contains(query.toLowerCase())) {
+                results.add(item);
+            }
+            else if (item instanceof PasswordVaultItem) {
+                final String usernameField = ((PasswordVaultItem) item).getUsername();
+                if (usernameField.toLowerCase().contains(query.toLowerCase())) {
+                    results.add(item);
+                }
+            }
```
Comment (HACKERALERT) [inline src/main/java/service/search/SearchInteractor.java:36]: Can you also search by url?

**Review by HACKERALERT — APPROVED** [review]

**Pull request merged**